htdocs/accountancy/admin/productaccount.php

```diff
@@ -612,7 +626,7 @@
 			// Accounting account buy export (Out of EEC)
 			print '<td class="left">';
 			//$defaultvalue=GETPOST('codeventil_' . $product_static->id,'alpha');        This is id and we need a code
-			if (empty($defaultvalue)) $defaultvalue = $compta_prodbuy_export;
+			if (empty($defaultvalue)) $defaultvalue = $compta_prodbuy;
```

Comment:
Why using the accounting account for local sale when selling out of europe and the the accounting account available for that ?

Reply:
This variable is unused. Whatever the situation, the variable $compta_prodbuy is used.

`		// Purchases
		if ($obj->product_type == 0) {
			if ($accounting_product_mode == 'ACCOUNTANCY_BUY') {
				**$compta_prodbuy** = (!empty($conf->global->ACCOUNTING_PRODUCT_BUY_ACCOUNT) ? $conf->global->ACCOUNTING_PRODUCT_BUY_ACCOUNT : $langs->trans("CodeNotDef"));
				$compta_prodbuy_id = $aarowid_prodbuy;
			} elseif ($accounting_product_mode == 'ACCOUNTANCY_BUY_INTRA') {
				**$compta_prodbuy** = (!empty($conf->global->ACCOUNTING_PRODUCT_BUY_INTRA_ACCOUNT) ? $conf->global->ACCOUNTING_PRODUCT_BUY_INTRA_ACCOUNT : $langs->trans("CodeNotDef"));
				$compta_prodbuy_id = $aarowid_prodbuy_intra;
			} elseif ($accounting_product_mode == 'ACCOUNTANCY_BUY_EXPORT') {
				**$compta_prodbuy** = (!empty($conf->global->ACCOUNTING_PRODUCT_BUY_EXPORT_ACCOUNT) ? $conf->global->ACCOUNTING_PRODUCT_BUY_EXPORT_ACCOUNT : $langs->trans("CodeNotDef"));
				$compta_prodbuy_id = $aarowid_prodbuy_export;
			} else {
				**$compta_prodbuy** = (!empty($conf->global->ACCOUNTING_PRODUCT_BUY_ACCOUNT) ? $conf->global->ACCOUNTING_PRODUCT_BUY_ACCOUNT : $langs->trans("CodeNotDef"));
				$compta_prodbuy_id = $aarowid_prodbuy;
			}
		} else {
			if ($accounting_product_mode == 'ACCOUNTANCY_BUY') {
				**$compta_prodbuy** = (!empty($conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT) ? $conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT : $langs->trans("CodeNotDef"));
				$compta_prodbuy_id = $aarowid_servbuy;
			} elseif ($accounting_product_mode == 'ACCOUNTANCY_BUY_INTRA') {
				**$compta_prodbuy** = (!empty($conf->global->ACCOUNTING_SERVICE_BUY_INTRA_ACCOUNT) ? $conf->global->ACCOUNTING_SERVICE_BUY_INTRA_ACCOUNT : $langs->trans("CodeNotDef"));
				$compta_prodbuy_id = $aarowid_servbuy_intra;
			} elseif ($accounting_product_mode == 'ACCOUNTANCY_BUY_EXPORT') {
				**$compta_prodbuy** = (!empty($conf->global->ACCOUNTING_SERVICE_BUY_EXPORT_ACCOUNT) ? $conf->global->ACCOUNTING_SERVICE_BUY_EXPORT_ACCOUNT : $langs->trans("CodeNotDef"));
				$compta_prodbuy_id = $aarowid_servbuy_export;
			} else {
				**$compta_prodbuy** = (!empty($conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT) ? $conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT : $langs->trans("CodeNotDef"));
				$compta_prodbuy_id = $aarowid_servbuy;
			}
		}`